specification/cognitiveservices/data-plane/ContentSafety/preview/2023-11-30-preview/contentsafety.json

```diff
@@ -0,0 +1,3161 @@
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ]
+    },
+    "Azure.Core.Foundations.ErrorResponse": {
+      "type": "object",
+      "description": "A response containing error details.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/Azure.Core.Foundations.Error",
+          "description": "The error object."
+        }
+      },
+      "required": [
+        "error"
+      ]
+    },
+    "Azure.Core.Foundations.InnerError": {
+      "type": "object",
+      "description": "An object containing more specific information about the error. As per Microsoft One API guidelines - https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses.",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "One of a server-defined set of error codes."
+        },
+        "innererror": {
+          "$ref": "#/definitions/Azure.Core.Foundations.InnerError",
+          "description": "Inner error."
+        }
+      }
+    },
+    "DetectUngroundednessGptxResource": {
+      "type": "object",
+      "description": "Connection information for GPT resource.",
+      "properties": {
+        "azureOpenAIEndpoint": {
+          "type": "string",
+          "description": "Endpoint of Azure OpenAI resource."
+        },
+        "deploymentName": {
+          "type": "string",
+          "description": "Deployment name of GPT model."
+        }
+      },
+      "required": [
+        "azureOpenAIEndpoint",
+        "deploymentName"
+      ]
+    },
+    "DetectUngroundednessOptions": {
+      "type": "object",
+      "description": "The request of ungroudedness detection.",
+      "properties": {
+        "domainCategory": {
+          "$ref": "#/definitions/TextDomainCategory",
+          "description": "Domain of the text to be analyzed."
+        },
+        "task": {
+          "$ref": "#/definitions/TextTask",
+          "description": "Task type of the text to be analyzed."
+        },
+        "qaQuery": {
+          "type": "string",
+          "description": "This is optional and only used for QnA or chat task."
+        },
+        "text": {
+          "type": "string",
+          "description": "The text needs to be analyzed. We support a maximum of 1k Unicode characters (Unicode code points) in the text of one request."
+        },
+        "groundingSources": {
+          "type": "array",
+          "description": "Source information that serves as grounding source. We support a maximum of 10k Unicode characters (Unicode code points) in all grounding sources of one request.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "gptxResource": {
+          "$ref": "#/definitions/DetectUngroundednessGptxResource",
+          "description": "GPT resources connection information for BYO scenario."
```

Comment:
> BYO

Let's try to avoid using abbreviation as much as possible unless it's well-known enough. Many customers may not understand BYO stands for bring your own.